app/bot.py

```diff
@@ -134,23 +134,69 @@ def send_file_info(self, result):
             self.respond("Hmm, I've found a resource here but can't parse it. Moving on...")
 
     def query(self, tokens):
+        
+        #Jas@20/1/16 - If a token has a - in front, it is removed and placed in filter_out
+        def get_filter_terms():
+            i = len(tokens)-1
+            filter_out = []
+            while i >= 0:  
+                if tokens[i][0] == '-':
+                    clean_token = tokens[i][1:]
+                    if len(clean_token) > 0:
+                        filter_out.append(clean_token)
+                        tokens.pop(i)
+                i -= 1
+            return filter_out
+        filter_out = get_filter_terms()
+        filter_terms_length = len(filter_out)
+ 
         """ Run a query
-        """        
+        """
+
         # Run the query
         query = '+'.join(tokens)
-        data = {'q': query, 'rows': 10}
+        
+        #Jas@20/1/16 -I made rows : 100, to give the user more details abot what's being filtered with search terms.
+        data = {'q': query, 'rows': 100} 
         query_response = requests.get(
             self.endpoint + '/package_search', params=data)
         
+        #Jas@20/1/16 - Loops through filter_out and then loops through the results['result']['results'] object from CKAN. If a str(results['result']['results']) contains the - term that result is removed
+        def filter_results_by_term(results, filter_out):
```

Comment:
generally you'd define a function like this outside the class in the base of the file, rather than right where you'd use it. Python programmers don't like breaking the flow of the script like JS ones do :)